### PR TITLE
Project updates

### PR DESCRIPTION
**Pull request by friederbluemle**
Various project updates. Best reviewed commit by commit.

It also fixes one accidentally misplaced `@DrawableRes` annotation.

<sub><sup>Please NO GitHub rebase/squash merges. If you'd like me to combine some or all of the commits, please let me know, and I will update my branch. Thank you.</sup></sub>

### CODE REVIEW

**Comment by PaulWoitaschek** [comment]
You could also update supportVersion to `23.3.1`

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Tue Feb 07 10:14:44 CST 2017
+#Sun Mar 05 15:33:01 SGT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-all.zip
```
Comment (egor-n) [inline gradle/wrapper/gradle-wrapper.properties]: 3.5 is the latest

Comment (friederbluemle) [inline gradle/wrapper/gradle-wrapper.properties]: Thanks @egor-n I amended the commit.

**Comment by friederbluemle** [comment]
@PaulWoitaschek Good point, I updated the support libs as well.

File: dependencies.gradle [hunk 2]
```diff
@@ -2,7 +2,7 @@ ext {
     minSdkVersion = 16
     compileSdkVersion = 25
     targetSdkVersion = 25
-    buildToolsVersion = '25.0.0'
+    buildToolsVersion = '25.0.2'
 
     supportVersion = '25.0.0'
```
Comment (PaulWoitaschek) [inline dependencies.gradle]: 25.3.1

**Comment by friederbluemle** [comment]
Rebased on latest `develop` branch.

**Comment by friederbluemle** [comment]
Rebased again, and build tools updated to `25.0.3`.

File: gradle.properties [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+# IDE (e.g. Android Studio) users:
+# Gradle settings configured through the IDE *will override*
+# any settings specified in this file.
+
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+org.gradle.jvmargs=-Xmx1536m
```
Comment (EricKuck) [inline gradle.properties]: I don't think we need to be specifying how much memory to allocate on people's machines. This belongs in a user's global gradle properties IMO.

Comment (friederbluemle) [inline gradle.properties]: @EricKuck Good catch, I was wondering about something similar when I first saw this. However, it turned out with more recent versions of the Android build toolchain, dex will run out of process (slowing down the build) if the maximum heap space is not set to at least 1536 MB. The jvmargs line I added is now **the default setting** when creating a new Android project.
Running Gradle *without* this parameter will output the following warning during the build:
```
Running dex as a separate process.

To run dex in process, the Gradle daemon needs a larger heap.
It currently has 1024 MB.
For faster builds, increase the maximum heap size for the Gradle daemon to at least 1536 MB.
To do this set org.gradle.jvmargs=-Xmx1536M in the project gradle.properties.
For more information see https://docs.gradle.org/current/userguide/build_environment.html
```
Note that `Xmx` only specifies the *maximum* memory allocation, not the initial amount (which would be `Xms`).

**Comment by friederbluemle** [comment]
Rebased, resolved conflicts, and updated to the latest version of the Android Gradle plugin (`2.3.2`).

**Comment by The0ldM0nk** [comment]
hey guys, any timeline as to when we can see this merged?

File: gradle.properties (hunk 3 above)
Comment (EricKuck) [inline gradle.properties]: Ah of course. Thanks for the clarification!

**Comment by friederbluemle** [comment]
@EricKuck Thanks for merging. Just a minor request: Next time, please let me know if you want me to squash some of the commits, instead of using GitHub squash merge. Thanks.

**Comment by EricKuck** [comment]
Sure I can do that - just curious why though? I've never heard anyone making that request before.

**Pull request merged**